### PR TITLE
Small `execution-core` cleanup

### PR DESCRIPTION
**Pull request by ureeves**
While working on moonlight, I had the urge to cleanup `execution-core` while working on it and thought it a good idea to make a separate PR.

Some functions are removed/changed, and some fields made public where they were otherwise private. In particular:

Changed:
- Transaction::new now takes a proof as impl Into<Vec<u8>>
- Transaction::proof now returns &[u8]
- Made Fee::stealth_address public
- Made Fee::sender public

Removed:
- Transaction::payload_hash
- Fee::sender
- Fee::stealth_address
- Payload::new
- Payload::tx_skeleton
- Payload::fee
- Payload::contract_call

### CODE REVIEW

**Review by HDauven — APPROVED** [review]
LGTM

File: contracts/transfer/src/state.rs [hunk 1]
```diff
@@ -190,9 +190,9 @@ impl TransferState {
             .extend_notes(block_height, tx_skeleton.outputs.clone());
 
         // if present, place the contract deposit on the state
-        if tx.payload().tx_skeleton().deposit > 0 {
-            let contract = match tx.payload().contract_call() {
-                Some(call) => ContractId::from_bytes(call.contract),
+        if tx.payload().tx_skeleton.deposit > 0 {
+            let contract = match &tx.payload().contract_call {
+                Some(call) => ContractId::from_bytes(call.contract.clone()),
```
Comment (moCello) [inline contracts/transfer/src/state.rs:195]: I think you don't even need to clone here

File: execution-core/src/transfer/transaction.rs [hunk 2]
```diff
@@ -272,7 +234,7 @@ impl Transaction {
     /// [`Sender::Encryption`].
     #[must_use]
     pub fn public_inputs(&self) -> Vec<BlsScalar> {
-        let tx_skeleton = self.payload().tx_skeleton();
+        let tx_skeleton = self.payload.tx_skeleton.clone();
```
Comment (moCello) [inline execution-core/src/transfer/transaction.rs:237]:
```suggestion
        let tx_skeleton = &self.payload.tx_skeleton;
```

**Review by moCello — DISMISSED** [review]
Two comments where a clone shouldn't be needed, for the rest LGTM

File: contracts/transfer/src/state.rs (hunk 1 above)
Comment (ureeves) [reply contracts/transfer/src/state.rs]: Fixed

File: execution-core/src/transfer/transaction.rs (hunk 2 above)
Comment (ureeves) [reply execution-core/src/transfer/transaction.rs]: Fixed

**Review by HDauven — APPROVED** [review]

**Pull request merged**